pokemon-checker/src/DataTransferObjects/PokemonDTO.ts

```diff
@@ -1,14 +1,50 @@
-type PokemonConstructorOptions = {
+import { MoveDTO } from "./MoveDTO";
+import { ITypeData, ISpriteData } from "../interfaces/PokemonData";
+
+export type PokemonConstructorOptions = {
   name: string;
   id: number;
-  types: { type: { name: string } }[];
-  moves: { name: string; url: string }[];
-  sprites: { front_shiny: string; front_default: string };
+  types: ITypeData[];
+  moves: MoveDTO[];
+  sprites: ISpriteData;
   stats: {
-    base_stat: number
-  }[]
+    base_stat: number;
+  }[];
 };
 
+export type pokemonDisplayObj = {
+  name?: string;
+  id?: string;
+  moves?: string[];
+  type1?: string;
+  type2?: string | null;
+  sprites?: {
+    frontDefault: string;
+    frontShiny: string;
+  };
+}
+
+export enum ElementType {
+  bug = "bug",
+  dark = "dark",
+  dragon = "dragon",
+  electric = "electric",
+  fairy = "fairy",
+  fighting = "fighting",
+  fire = "fire",
+  flying = "flying",
+  ghost = "ghost",
+  grass = "grass",
+  ground = "ground",
+  ice = "ice",
+  normal = "normal",
+  poison = "poison",
+  psychic = "psychic",
+  rock = "rock",
+  steel = "steel",
+  water = "water",
```

Comment:
I'm wondering if the enum names should be capitalized
ex: BUG = "bug",

Reply:
That's a matter of style, but I think all caps enums looks tight so let's do that.